pyproject.toml

```diff
@@ -40,13 +40,13 @@ qiskit-aer = "^0.13.0"
 lark = "~=0.12"
 regex = "^2022.6.2"
 jsonpickle = "^2.2.0"
+black = "24.2.0"
```

Comment:
Poetry install; Poetry run black ., doesn't interfer with development, Poetry run black needs to be run regardless. 
This also doesn't interfer with other projects as if they're using it directly it's pip install and if it's a git sub project that's independent of the parent project. 

Updating dependencies in dev should never be an issue

Reply:
I mean if you put it in the normal dependencies it states our project requires it for execution, it gets imported as a dependency when other people install it via pypi. So if someone does pip install qat-compiler it will also install Black, which may mess with their own environment. You can't assume what sort of environment other people run with, as there are some pretty whacky setups around. (And if you're suggesting we have different toml files between develop/main/deploy that's not really needed here)

And all that aside it's _not_ required for execution, so its just a redundant dependancy and bloats things when people install.

Reply:
Oh, completely agree, shouldn't be in the normal dependencies, entirely missed that in poetry upgrading. Thought you were referring to just the version change, will rip out of main dependencies